opencensus/proto/metrics/metrics.proto

```diff
@@ -0,0 +1,209 @@
+// Copyright 2016-18, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package opencensus.proto.metrics;
+
+import "google/protobuf/timestamp.proto";
+
+option go_package = "metricsproto";
+option java_multiple_files = true;
+option java_package = "io.opencensus.proto.metrics";
+option java_outer_classname = "MetricsProto";
+
+// A collection of MetricPoint's.
+message MetricPointSet {
+  // Each MetricPoint has one or more timeseries for a single metric.
+  repeated MetricPoint metric_points = 1;
+}
+
+// Defines a MetricPoint which has one or more timeseries for a single metric.
+message MetricPoint {
+  // The description of the metric.
+  MetricDescriptor desctriptor = 1;
+
+  // One or more timeseries for a single metric, where each timeseries has
+  // one or more points.
+  repeated TimeSeries timeseries = 2;
+}
+
+// Defines a metric type and its schema.
+message MetricDescriptor {
+  // The metric type, including its DNS name prefix. It must be unique.
+  string name = 1;
+
+  // A detailed description of the metric, which can be used in documentation.
+  string description = 2;
+
+  // The unit in which the metric value is reported. Follows the format
+  // described by http://unitsofmeasure.org/ucum.html.
+  string unit = 3;
+
+  // The kind of metric. It describes how the data is reported.
+  enum Type {
+    // Do not use this default value.
+    UNSPECIFIED = 0;
+
+    // An instantaneous measurement of a value.
+    GAUGE = 1;
+
+    // A value accumulated over a time interval. Cumulative measurements in a
+    // time series should have the same start time and increasing end times,
+    // until an event resets the cumulative value to zero and sets a new
+    // start time for the following points.
+    CUMULATIVE = 2;
+  }
+  Type type = 4;
+
+  // TODO: Consider to add LabelInfo{key, description} if needed.
+}
+
+// A collection of data points that describes the time-varying values
+// of a metric.
+message TimeSeries {
+  // The set of label values that uniquely identify this metric. Apply to all
+  // points.
+  repeated Label label = 1;
+
+  // The data points of this timeseries.
+  repeated Point points = 2;
+}
+
+message Label {
+  // The label name.
```

Comment:
Similarly, should we mention the constraints here?

Reply:
Similar, added a TODO for the moment.